WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsV4ChartAxisHelper.swift

```diff
@@ -0,0 +1,51 @@
+import Yosemite
+
+/// A helper for the labels for store stats v4 charts UI.
+class StoreStatsV4ChartAxisHelper {
+    func labelCount(timeRange: StatsTimeRangeV4) -> Int {
+        let labelCount: Int
+        switch timeRange {
+        case .thisYear:
+            labelCount = 4
+        case .today:
+            labelCount = 5
+        case .thisMonth:
+            labelCount = 6
+        case .thisWeek:
+            labelCount = 7
+        }
+        return labelCount
+    }
+
+    /// Generates an array of label text that is shown on the chart x axis
+    ///
+    /// - Parameter dates: an array of date for each stats interval
+    /// - Returns: an array of the same length as input, that contains the text to be shown on chart x axis
+    func generateLabelText(for dates: [Date], timeRange: StatsTimeRangeV4, siteTimezone: TimeZone) -> [String] {
+        let chartDateFormatter = timeRange.chartDateFormatter(siteTimezone: siteTimezone)
+        guard timeRange.intervalGranularity == .daily else {
+            return dates.map({ chartDateFormatter.string(from: $0) })
+        }
+
+        var dateLabelText = [String]()
+
+        var calendar = Calendar.current
+        calendar.timeZone = siteTimezone
+        var latestMonth: Int? = nil
+        let dayOfMonthDateFormatter = DateFormatter.Charts.chartAxisDayOfMonthFormatter
+        dayOfMonthDateFormatter.timeZone = siteTimezone
+        for date in dates {
```

Comment:
Just out of curiosity, and for kicks and giggles, I tried to write this using reduce, and it was way more compact, but also way more unreadable (at least to me)

Reply:
I like reduce! went for the most straightforward for loop first, lemme rewrite with reduce